Basic/Basic/AppDelegate.m

```diff
@@ -9,41 +9,40 @@
 #import "AppDelegate.h"
 #import "AppDelegate.h"
 #import "MainViewController.h"
-#import "Employee.h"
-#import "Organization.h"
 #include "stdlib.h"
 
-@interface AppDelegate ()
+@interface AppDelegate () <UINavigationControllerDelegate>
 
 @end
 
 @implementation AppDelegate
 
++(AppDelegate *)instance
+{
+    return (AppDelegate *)[UIApplication sharedApplication].delegate;
+}
 
-- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-    // Override point for customization after application launch.
-        
-    Employee *empl = [[Employee alloc] initWithFirstName:@"Dima" lastName:@"Bondar" salary:1200];
-    
-    NSLog(@"Full name: %@", [empl fullName]);
-    
-    Organization * organ = [[Organization alloc]initWithName:@"ww ee"];
-    
-    [organ addEmployeeWithName:@"tt rr"];
-    [organ addEmployeeWithName:@"tw rw"];
-    [organ addEmployeeWithName:@"tc rc"];
-    [organ addEmployee:empl];
-    
-    NSLog(@"Avarage salary: %f", [organ calculateAverageSalary]);
-    NSLog(@"LowestSalary: %i", [organ employeeWithLowestSalary]);
-    for (Employee * em in [organ employeesWithSalary:1300 tolerance:200]) {
-        NSLog(@"Tolerance: %i", em.salary);
+
+- (NSPersistentContainer *)persistentContainer
+{
+    if (_persistentContainer == nil) {
```

Comment:
In objective-c to check if the object exists, use `if (!_persistentContainer)` format.